services/components.py

```diff
@@ -80,42 +80,50 @@ def patch_totals(self) -> ReportTotals:
 class ComponentMeasurements:
     def __init__(
         self,
-        raw_measurements: Iterable[dict],
+        raw_measurements: List[dict],
         component_id: str,
         interval: Interval,
         after: datetime,
         before: datetime,
         last_measurement: datetime,
+        components_mapping: Dict[str, str],
     ):
         self.raw_measurements = raw_measurements
         self.component_id = component_id
         self.interval = interval
         self.after = after
         self.before = before
         self.last_measurement = last_measurement
+        self.components_mapping = components_mapping
 
     @cached_property
-    def name(self):
+    def name(self) -> str:
+        if self.components_mapping.get(self.component_id):
```

Comment:
Not really because the `shared.Components` object would always set the `name` to be empty string if it is not defined (never None). So we would have dicts like:
```
components_mapping = {
  "Id1": "",
  "Id2": "Name1",
}
```
using the `.get` would return the empty string as name now because it found the empty string name value for the ID, but ofc doing `if ""` returns False